packages/client/src/index.ts

```diff
@@ -43,19 +53,68 @@ export const createClient = <schema extends Schema>(
   url: string,
   { schema }: { schema: schema },
 ): Client<schema> => {
-  const db = drizzle(
-    async (sql, params, method, typings) => {
-      const result = await fetch(`${url}/client`, {
-        method: "POST",
-        body: JSON.stringify({ sql, params, method, typings }),
-      });
-
-      return await result.json();
+  const noopDatabase = drizzle(
+    () => {
+      return Promise.resolve({ rows: [] });
     },
     { schema, casing: "snake_case" },
   );
 
-  return { db };
+  // @ts-ignore
+  const dialect: PgDialect = noopDatabase.dialect;
+
+  // @ts-ignore
+  const session: PgSession = noopDatabase.session;
+
+  return {
+    db: drizzle(
+      async (sql, params, method, typings) => {
+        const result = await fetch(`${url}/client/db`, {
+          method: "POST",
+          body: JSON.stringify({ sql, params, method, typings }),
+        });
+
+        return await result.json();
+      },
+      { schema, casing: "snake_case" },
+    ),
+    live: (_query, callback) => {
+      // https://github.com/drizzle-team/drizzle-orm/blob/04c91434c7ac10aeb2923efd1d19a7ebf10ea9d4/drizzle-orm/src/pg-core/db.ts#L602-L621
+
+      const query = _query(noopDatabase) as unknown as SQLWrapper | string;
+      const sequel =
+        typeof query === "string" ? sql.raw(query) : query.getSQL();
+      const builtQuery = dialect.sqlToQuery(sequel);
+
+      const prepared = session.prepareQuery(
+        builtQuery,
+        undefined,
+        undefined,
+        false,
+      );
+
+      const sse = new EventSource(
+        `${url}/client/live?${new URLSearchParams({
+          query: JSON.stringify(builtQuery),
+        }).toString()}`,
+      );
+
+      sse.onmessage = (event) => {
+        // @ts-ignore
+        callback(prepared.mapResult(JSON.parse(event.data), true).rows);
+      };
+
+      sse.onerror = () => {
```

Comment:
Perhaps the API should have an onError callback? Does the EventSource API include the event object here?